recipe/build.sh

```diff
@@ -23,10 +23,10 @@ ${BUILD_PREFIX}/bin/cmake ${CMAKE_ARGS} \
   -D BUILD_SHARED_LIBS=ON \
   -D Eigen3_ROOT=${PREFIX} \
   -D ENABLE_XHOST=OFF \
-  -D LIBINT2_REQUIRE_CXX_API=ON \
-  -D LIBINT2_REQUIRE_CXX_API_COMPILED=OFF \
-  -D LIBINT2_ENABLE_FORTRAN=OFF \
-  -D LIBINT2_ENABLE_PYTHON=OFF \
+  -D REQUIRE_CXX_API=ON \
+  -D REQUIRE_CXX_API_COMPILED=OFF \
+  -D ENABLE_FORTRAN=ON \
+  -D ENABLE_PYTHON=OFF \
```

Comment:
cmake says that it does not know about those variables with LIBINT2 prefix. I removed the prefix according to some old docs from 2016.
When removing them, cmake knows about REQUIRE_CXX_API and ENABLE_FORTRAN.
ENABLE_FORTRAN=ON also generates the fortran module.

ENABLE_PYTHON and REQUIRE_CSS_API_COMPILED are still unknown. I left them here however.

Reply:
Build of CP2K fails with:
```
Error: Symbol 'libint2_init_2eri' referenced at (1) not found in module 'libint_f'
```
and
```
 ERROR: __LIBINT_MAX_AM is too large
```

Reply:
The latter error is most probably caused by the toy source. I will let a full build run over night. Do you have insight into the former @loriab ?

Reply:
Ah just checked the toy build config and eri2 is off. Will try with the full build

Reply:
* Yeah, don't worry about the unrecognized variables. I had left the names as the new, regularized cmake names, but the on/off settings were the defaults, so it didn't matter until you enabled Fortran.
* Right, the toy src is enough to build that Hartree--Fock test and not much more.
* Let me know if the full build was enough to get cp2k linked, please. I started the split-packages recipe, then looked at the linux artifact for your latest build and realized that there isn't a fortran library, only the `include/libint_f.mod` file is new.
  From the src, this is expected as only the module file is installed: https://github.com/evaleev/libint/blob/master/export/cmake/CMakeLists.txt.export#L559 . So perhaps this is all cp2k needs? If so, I was wrong about needing split packages and you just need a `# [unix]` selector on the fortran compiler in meta.yaml to "fix" windows. Or maybe more needs to be installed for the fortran build -- the cmake builds an object file but no library.

Reply:
Yes, the only changed file appears to be the module. This would not be a problem. The larger change is the produced meta.yaml in the fortran build:
```
Without gfortran compiler dep (from your compiled package):
  run:
    - libgcc-ng >=12
    - libstdcxx-ng >=12
With the fortran dep (from my compiled package:
  run:
    - libgcc-ng >=12
    - libgfortran-ng
    - libgfortran5 >=12.3.0
    - libstdcxx-ng >=12
```
In other words: We incur a gfortran dependency for pure C++ users, which is not required for c++ users. The solution could be one build step with two subpackages, where the c++ package uses:
```
build:
  ignore_run_exports_from:
    - {{ compiler('fortran') }}
```
and depends on the c++ package and contains only the module file.

I didn't try the build with CP2K yet, but will soon.

Reply:
Tests passed on the CP2K side, so ENABLE_FORTRAN seems to be the only thing required. I also disabled build for Windows. If we decide to ignore the added fortran library run dependency, this would be good to go, once I remove the two bogus commits and rerender, which I will do now.